### ISSUE TITLE
Paths aren't working properly/aren't clear when using npm 8

### ISSUE THREAD

**Issue by leedm777** [IBM/audit-ci#213]
We are working through updating our projects to Node.js v16, and npm v8. We've discovered that the paths in the allowlist don't work the say they did when using npm v6.

Here is an example config we are using that works fine with npm v6, installing swagger-stats@0.99.2:

```
{
  "low": true,
  "allowlist": ["1006724|swagger-stats>request>http-signature>jsprim>json-schema"],
  "pass-enoaudit": true
}
```

After much poking and reverse engineering, I've discovered that each step of the path needs its own entry in the allowlist to be properly excluded:

```
{
  "low": true,
  "allowlist": [
    "1006724|http-signature>json-schema",
    "1006724|jsprim>json-schema",
    "1006724|request>json-schema",
    "1006724|swagger-stats>json-schema"
  ],
  "pass-enoaudit": true
}
```

This is in spite of the fact that there is only one install of `json-schema` in the tree:

```
❯ npm ls json-schema
ex@1.0.0 /Users/dlee/prj/ex
└─┬ swagger-stats@0.99.2
  └─┬ request@2.88.2
    └─┬ http-signature@1.2.0
      └─┬ jsprim@1.4.1
        └── json-schema@0.2.3
```

Here's an example output from `npm audit --json` for our project when using npm v8:
```
{
  "auditReportVersion": 2,
  "vulnerabilities": {
    "http-signature": {
      "name": "http-signature",
      "severity": "moderate",
      "isDirect": false,
      "via": [
        "jsprim"
      ],
      "effects": [
        "request"
      ],
      "range": "1.0.0 - 1.3.5",
      "nodes": [
        "node_modules/http-signature"
      ],
      "fixAvailable": {
        "name": "swagger-stats",
        "version": "0.94.1",
        "isSemVerMajor": true
      }
    },
    "json-schema": {
      "name": "json-schema",
      "severity": "moderate",
      "isDirect": false,
      "via": [
        {
          "source": 1006724,
          "name": "json-schema",
          "dependency": "json-schema",
          "title": "json-schema is vulnerable to Prototype Pollution",
          "url": "https://github.com/advisories/GHSA-896r-f27r-55mw",
          "severity": "moderate",
          "range": "<0.4.0"
        }
      ],
      "effects": [
        "jsprim"
      ],
      "range": "<0.4.0",
      "nodes": [
        "node_modules/json-schema"
      ],
      "fixAvailable": {
        "name": "swagger-stats",
        "version": "0.94.1",
        "isSemVerMajor": true
      }
    },
    "jsprim": {
      "name": "jsprim",
      "severity": "moderate",
      "isDirect": false,
      "via": [
        "json-schema"
      ],
      "effects": [
        "http-signature"
      ],
      "range": "0.3.0 - 2.0.1",
      "nodes": [
        "node_modules/jsprim"
      ],
      "fixAvailable": {
        "name": "swagger-stats",
        "version": "0.94.1",
        "isSemVerMajor": true
      }
    },
    "request": {
      "name": "request",
      "severity": "moderate",
      "isDirect": false,
      "via": [
        "http-signature"
      ],
      "effects": [
        "swagger-stats"
      ],
      "range": ">=2.66.0",
      "nodes": [
        "node_modules/request"
      ],
      "fixAvailable": {
        "name": "swagger-stats",
        "version": "0.94.1",
        "isSemVerMajor": true
      }
    },
    "swagger-stats": {
      "name": "swagger-stats",
      "severity": "moderate",
      "isDirect": true,
      "via": [
        "request"
      ],
      "effects": [],
      "range": ">=0.95.0",
      "nodes": [
        "node_modules/swagger-stats"
      ],
      "fixAvailable": {
        "name": "swagger-stats",
        "version": "0.94.1",
        "isSemVerMajor": true
      }
    }
  },
  "metadata": {
    "vulnerabilities": {
      "info": 0,
      "low": 0,
      "moderate": 5,
      "high": 0,
      "critical": 0,
      "total": 5
    },
    "dependencies": {
      "prod": 308,
      "dev": 847,
      "optional": 10,
      "peer": 0,
      "peerOptional": 0,
      "total": 1162
    }
  }
}
```

**Comment by arokor** [comment]
This issue is probably important to a lot of people since npm 8 comes with node 16 which is currently the latest LTS version. Is npm 8 support in the pipeline?

**Comment by quinnturner** [comment]
I agree that the current state of NPM 8's path resolution is an "incorrect" implementation. Notably, the new audit format makes it very challenging to map to the same result. I am actively working on resolving this issue now and bundling it within #219.

**Comment by quinnturner** [comment]
Resolved with [6.0.0-beta.0](https://www.npmjs.com/package/audit-ci/v/6.0.0-beta.0). The path resolution algorithm should work identically to NPM 6. _Please_, provide feedback if it works for you or any issues that come up!

**Comment by quinnturner** [comment]
Released `6.0.0`
